packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx

```diff
@@ -110,94 +95,156 @@ export function CreateForm(props: CreateFormProps) {
-            return 'Enter an amount'
+        if (!account) {
+            return t('plugin_wallet_connect_a_wallet')
         }
-        if (!token) {
-            return 'Select to token'
+        if (!tokenAndAmounts || tokenAndAmounts.length === 0) {
+            return t('plugin_ito_error_enter_amount_and_token')
         }
-        return ''
-    }, [token, amount])
+        for (const { amount, token } of tokenAndAmounts) {
+            if (!token) {
+                return t('plugin_ito_error_select_token')
+            }
 
-    const onTotalOfPerWalletChange = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
-        const total = ev.currentTarget.value.replace(/[,\.]/g, '')
-        if (total === '') setTotalOfPerWallet(0)
-        else if (/^[1-9]+\d*$/.test(total)) {
-            setTotalOfPerWallet(Number.parseInt(total, 10))
+            if (new BigNumber(amount).isZero()) {
+                return t('plugin_ito_error_enter_amount')
+            }
         }
-    }, [])
 
-    const onDaysChange = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
-        const days_ = ev.currentTarget.value.replace(/[,\.]/g, '')
-        console.log(days_)
-        if (days_ === '') setDays(0)
-        else if (/^[0-9]+\d*$/.test(days_)) {
-            setDays(Number.parseInt(days_, 10))
+        if (new BigNumber(tokenAndAmount?.amount ?? '0').isGreaterThan(new BigNumber(tokenBalance)))
+            return `Insufficient ${tokenAndAmount?.token?.symbol} balance`
+
+        if (Number(totalOfPerWallet) === 0) {
+            return t('plugin_ito_error_enter_allocation_per_walllet')
         }
-    }, [])
-    const onHoursChange = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
-        const hours_ = ev.currentTarget.value.replace(/[,\.]/g, '')
-        if (hours_ === '') setHours(0)
-        else if (/^[0-9]+\d*$/.test(hours_)) {
-            setHours(Number.parseInt(hours_, 10))
+
+        if (startTime && endTime && startTime >= endTime) {
+            return t('plugin_ito_error_exchange_time')
         }
-    }, [])
 
-    useEffect(() => {
-        console.log('**********************')
-        console.log(amount)
-        console.log(token)
-        console.log(tokenAndAmount)
-        console.log(message)
-        console.log(totalOfPerWallet)
-    }, [amount, token, tokenAndAmount, message, totalOfPerWallet])
+        return ''
+    }, [
+        account,
+        endTime,
+        startTime,
+        t,
+        tokenAndAmount?.amount,
+        tokenAndAmount?.token?.symbol,
+        tokenAndAmounts,
+        tokenBalance,
+        totalOfPerWallet,
+    ])
+
+    const handleStartTime = useCallback(
+        (timeString) => {
+            const now = new Date()
+            if (endTime === '' && new Date(timeString).getTime() >= now.getTime()) {
+                setStartTime(timeString)
+            }
+            if (
+                new Date(timeString).getTime() < new Date(endTime).getTime() &&
```

Comment:
why not store `new Date(timeString).getTime()` result?

Reply:
Limit the range of date input